design/rook-ceph-status-conditions.md

```diff
@@ -0,0 +1,162 @@
+`Reason` is intended to be a one-word, CamelCase representation of the category of cause of the current status, and `Message` is intended to be a human-readable phrase or sentence, which may contain specific details of the individual occurrence. `Reason` is intended to be used in concise output, such as one-line kubectl get output, and in summarizing occurrences of causes, whereas `Message` is intended to be presented to users in detailed status explanations, such as `kubectl describe output`.
+
+In the CephClusterStatus, we can either remove the `Status.State` and `Status.Message` fields and call the `Conditions` structure from inside the `ClusterStatus`, or we can just add the `Conditions` structure by keeping the already included fields.The first method is preferred because the `Conditions` structure contains the `Conditions.Type` and `Conditions.Message` which is similar to the `Status.State` and `Status.Message`. According to the above changes, necessary changes are to be made everywhere `ClusterStatus` or one of its fields are referred. 
+
+
+
+### Examples
+
+Consider a cluster is being created. The RookConditions is an array that can store multiple Conditions. So the progression of the cluster being created can be seen in the RookConditions as shown in the example below. The Ceph Cluster gets created after it establishes a succesfull Connection. The `RookCondition` will show in the slice that the `Connecting` Condition will be in `Condition.Status` False. The `Connected` and `Progressing` Types will be set to True.
+
+	   Before:
+		ClusterStatus{
+		    State      :   Creating,
+	   	    Message    :   The Cluster is getting created,
+		    CephStatus :   *CephStatus,
+		}
+	   After:
+		ClusterStatus{
+		     RookConditions{
+			{
+			  Type    :   Connecting,
+			  Status  :   False,
+			  Reason  :   ClusterConnecting,
+			  Message :   The Cluster is Connecting,
+			},
+			{
+			  Type    :   Connected,
+			  Status  :   True,
+			  Reason  :   ClusterConnected,
+			  Message :   The Cluster is Connected,
+			}
+			{
+			  Type    :   Progressing,
+			  Status  :   True,
+			  Reason  :   ClusterCreating,
+			  Message :   The Cluster is getting created,
+			},
+		     },
+		     CephStatus :   *CephStatus,
+		}
+During the Updation of a Cluster, the `Progressing`, `Ready` and `Available` will be set to False. The `ConditionType` `NotReady` will be set to True and wait for the Cluster to finish its Updation. Once the updation completes, the Reconcilation part will start again and get the Cluster into the `Available` Condition. 
+
+
+	   Before:
+		ClusterStatus{
+		    State      :   Updating,
+	   	    Message    :   The Cluster is getting updated,
+		    CephStatus :   *CephStatus,
+		}
+	   After:
+		ClusterStatus{
```

Comment:
Just to be clear, all of these statuses will always exist in the CR, right? They will just toggle between true and false depending on what the operator is doing? So the conditions are very verbose and hard to read for a human. Is there a readable status somewhere? A readable status [here](https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/ceph/common.yaml#L196-L199) is helpful.

Reply:
+1 for readable status.
Part of it is already touched here, https://github.com/rook/rook/pull/4066#discussion_r334758459

Reply:
> Just to be clear, all of these statuses will always exist in the CR, right? They will just toggle between true and false depending on what the operator is doing?
Yes. All of the condition will exist in the CR and depending on what the operator is doing, the status of the ConditionType will be set to True or False. 




> A readable status [here](https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/ceph/common.yaml#L196-L199) is helpful.

Yes. It is helpful. I will add that too.